teos-common/src/lib.rs

```diff
@@ -79,10 +80,111 @@ impl TryFrom<serde_json::Value> for UserId {
                     ))
                 }
             }
+            serde_json::Value::Object(mut m) => {
+                let param_count = m.len();
+                if param_count > 1 {
+                    Err(format!(
+                        "Unexpected json format. Expected a single parameter. Received: {}",
+                        param_count
+                    ))
+                } else {
+                    UserId::try_from(json!(m
+                        .remove("user_id")
+                        .or_else(|| m.remove("tower_id"))
+                        .ok_or("user_id or tower_id not found")?))
+                }
+            }
```

Comment:
Just to asses my understanding, this part is actually never triggered, correct?

Reply:
It is actually. It depends on how the params are passed. By default, cln passes the params as an `Array` of json values, but if `-k` is specified, then that's a json `Object` which translates to a `Map`.

Reply:
Yes but if the params are passed as `-k` then the `RegisterParams` or `GetAppointmentParams` will convert `tower_id` to a string before hand and pass the whole thing as an array just like the older positional parameters version. So this piece won't get triggered.

Reply:
My bad, this is actually triggered in other methods which require `tower_id` other than `register` and `getappointment`, such as `getregistrationreceipt` and `gettowerinfo`.

Reply:
Umm, still you made me realize that the implementation may be wrong tho. The `Map` branch checks that the provided keys are correct, but puts the elements in the received order. If params are passed in any other order than the expected for the non `-k` alternative, then I'm pretty sure this will fail :(

Reply:
Nvm, we are looping over `allowed_keys` in the proper order, so the list will always be properly ordered